File: Android project/RoomDataBase/app/src/main/java/com/mirkamol/roomdatabase/manager/RoomManager.java
package com.mirkamol.roomdatabase.manager;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.mirkamol.roomdatabase.database.UserDao;
import com.mirkamol.roomdatabase.model.User;


@Database(entities = {User.class}, version = 1)
public abstract class RoomManager extends RoomDatabase {

    public abstract UserDao usersDao();
    private static RoomManager INSTANCE;

    public static RoomManager getDatabase(final Context context){
        if (INSTANCE == null){
            synchronized (RoomManager.class){
                if (INSTANCE == null){
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                    RoomManager.class, "app_db")
                            .fallbackToDestructiveMigration()
                            .build();
                }
            }
        }

        return INSTANCE;
    }
}

File: Android project/SavedtoRoomfromServer/app/src/main/java/com/mirkamol/savedtoroomfromserver/manager/RoomManager.kt
package com.mirkamol.savedtoroomfromserver.manager

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.mirkamol.savedtoroomfromserver.database.UserDao
import com.mirkamol.savedtoroomfromserver.model.UserItem

@Database(entities = [UserItem::class], version = 1)
abstract class RoomManager:RoomDatabase() {
    abstract fun userDao(): UserDao

    companion object{
        private var INSTANCE:RoomManager? = null

        fun getDatabase(context: Context):RoomManager?{
            if (INSTANCE == null){
                synchronized(RoomManager::class.java){
                    INSTANCE = Room.databaseBuilder(context.applicationContext, RoomManager::class.java, "users.db")
                        .fallbackToDestructiveMigration()
                        .allowMainThreadQueries()
                        .build()
                }
            }
            return INSTANCE
        }
    }

}